client_sdk/go/fpc/packager.go

```diff
@@ -0,0 +1,204 @@
+package fpc
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/hyperledger/fabric/core/chaincode/persistence"
+	"github.com/hyperledger/fabric/core/chaincode/platforms/util"
+	"github.com/pkg/errors"
+)
+
+/*
+Notice: This file is has been modified for Fabric Private Chaincode.
+*/
+
+const (
+	codePackageName      = "code.tar.gz"
+	metadataPackageName  = "metadata.json"
+	mrenclaveFileName    = "mrenclave"
+	enclaveBinaryName    = "enclave.signed.so"
+	gzipCompressionLevel = gzip.DefaultCompression
+	sgxModeEnvKey        = "SGX_MODE"
+	ChaincodeLang        = "fpc-c"
+)
+
+// NewCCPackage creates a fpc chaincode package
+func NewCCPackage(desc *Descriptor) ([]byte, error) {
+	err := desc.Validate()
+	if err != nil {
+		return nil, err
+	}
+
+	pkgTarGzBytes, err := getTarGzBytes(desc, writePackage)
+	if err != nil {
+		return nil, err
+	}
+
+	return pkgTarGzBytes, nil
+}
+
+// Descriptor holds the package data
+type Descriptor struct {
+	Path  string
+	Type  string
+	Label string
+}
+
+// ReadMrenclave returns mrenclave for the given chaincode at ccPath
+func ReadMrenclave(ccPath string) (string, error) {
```

Comment:
Is this (and below `Validate`) intentionally a public function?  For a public API i think we should minimize what is exposed ...

Reply:
`ReadMrenclave` should be a public function as it is used by the sdk users to get the mrenclave for the other lifecycle operations.

`Validate` can be private. In the go sdk it is also public but only used within its package.

Reply:
ah, ok.